File: charts/src/services/candles.js
import axios from 'axios'
import config from '../config.json'

export const fetchCandles = (
  timeframe,
  start,
  end
) => {
  return axios.get(`${config.base_url}${config.candles.path}?timeframe=${timeframe}&start=${start}&end=${end}`).then(response => {
    // Format candle data
    return response.data.data.map(element => {
      return {
        time: element[0] / 1000,
        time_ms: element[0],
        open: element[1],
        close: element[2],
        high: element[3],
        low: element[4],
      }
    })
})
  .catch(error => 
    console.log(error)
  )
}

File: prod_app/pipeline/candles.py
import requests

base_url = "https://api-pub.bitfinex.com/v2/candles/trade:%s:t%s/%s"

def get_current_candle(symbol, timeframe):
    url = base_url % (timeframe, symbol.upper(), "last")
    candles_data = requests.get(url).json()
    candle = {
        'timestamp': candles_data[0],
        'open': candles_data[1],
        'close': candles_data[2],
        'high': candles_data[3],
        'low': candles_data[4],
    }
    return candle
